sharding/proposer/p2p_test.go

```diff
@@ -0,0 +1,159 @@
+package proposer
+
+import (
+	"bytes"
+	"errors"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/sharding"
+	"github.com/ethereum/go-ethereum/sharding/p2p"
+	"github.com/ethereum/go-ethereum/sharding/params"
+)
+
+type faultyRequest struct{}
+type faultySigner struct{}
+type faultyCollationFetcher struct{}
+
+type mockSigner struct{}
+type mockCollationFetcher struct{}
+
+func (m *mockSigner) Sign(hash common.Hash) ([]byte, error) {
+	return []byte{}, nil
+}
+
+func (f *faultySigner) Sign(hash common.Hash) ([]byte, error) {
+	return []byte{}, errors.New("could not sign hash")
+}
+
+func (m *mockCollationFetcher) CollationByHeaderHash(headerHash *common.Hash) (*sharding.Collation, error) {
+	shardID := big.NewInt(1)
```

Comment:
Hey constants like `shardID` and `period` can be declared on top as global vars will be easier to simply use the var then